### DISCUSSION TITLE
How can I upgrade my crashlytics SDK without migrating my current react-native-firebse/crashlytics which is in 6.3.4

### DISCUSSION THREAD

**Discussion by amitmundrag08** [invertase/react-native-firebase#4435] [Q&A] [answered]
When I upgrade all the libraries I get issue while fetching data. Currently I am using :-
         Current                                                                                        Upgrading to
  "@react-native-firebase/analytics": "6.3.4",		     "@react-native-firebase/analytics": "7.6.7",
     "@react-native-firebase/app": "6.3.4",		     "@react-native-firebase/app": "8.4.5",
     "@react-native-firebase/auth": "6.3.4",		     "@react-native-firebase/auth": "9.3.0",
     "@react-native-firebase/crashlytics": "6.3.4",		     "@react-native-firebase/crashlytics": "8.4.9",
     "@react-native-firebase/dynamic-links": "6.3.4",		     "@react-native-firebase/dynamic-links": "7.5.9",
     "@react-native-firebase/firestore": "6.3.4",		     "@react-native-firebase/firestore": "7.8.6",
     "@react-native-firebase/functions": "6.3.4",		     "@react-native-firebase/functions": "7.4.8",
     "@react-native-firebase/messaging": "6.3.4",		     "@react-native-firebase/messaging": "7.9.0",
     "@react-native-firebase/perf": "6.3.4",		     "@react-native-firebase/perf": "7.4.8",
     "@react-native-firebase/storage": "6.3.4",		     "@react-native-firebase/storage": "7.4.9",

But when I updated I am getting issues while fetching data from server (firestore). Please suggest me something. This issue is specifically for android. iOS everything seems to work fine.

Sharing a sample of code :- 

Here  "lastSyncedServerTimestamp" is the last time when I fetched data from server 

firestore().collection("CollectionName").where(
  "_serverDocumentUpdatedOn",
  ">",
  new Date(lastSyncedServerTimestamp.valueOf() - 5 * 60000)
) // 5 minutes before the last refresh serverDateTimeStamp to provide for server side time overlap and misses
.orderBy("_serverDocumentUpdatedOn", "desc")
.onSnapshot((change) => {

})

Also I am using firestore.FieldValue.serverTimestamp() as _serverDocumentUpdatedOn.

**Comment by mikehardy** [comment]
Sorry, I'm not sure what "issue" you are having specifically so I can't recommend anything. In general we recommend using current stable versions of everything though

**Reply by amitmundrag08** [reply]
I can elaborate my issue. I am trying to update data and while retrieving data on method onSnapshot i'm getting change.type as 'removed' instead of 'modified' for android

**Reply by amitmundrag08** [reply]
Please recmommend me a stable library version which I can use to upgrade my crashlytics library

**Reply by amitmundrag08** [reply]
On which library do we depend to get data from native code to js code in the above said libraries.

**Reply by mikehardy** [reply]
Use current stable versions

**Reply by amitmundrag08** [reply]
How to identify the current stable version?

**Comment by andersonaddo** [comment, marked as answer]
https://rnfirebase.io/releases lists all our releases

If you want to upgrade the SDKs independently of react-native-firebase:
Latest Android versions: https://firebase.google.com/support/release-notes/android#bom_v26-0-0
Latest iOS versions: https://firebase.google.com/support/release-notes/ios#version_700_-_october_26_2020

And here's a PR that shows how to manually upgrade the native SDKs:
https://github.com/invertase/react-native-firebase/pull/3881
